cmake/modules/Utility.cmake

```diff
@@ -206,19 +201,20 @@ function(add_swift_module target name deps sources additional_args)
   endforeach()
 
   # Add rpath to lookup the linked dylibs adjacent to itself.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
     list(APPEND DYLYB_ARGS -Xlinker -rpath -Xlinker @loader_path)
-    list(APPEND DYLYB_ARGS -Xlinker -install_name -Xlinker @rpath/${target}.${DYLIB_EXT})
-  else()
-    list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN")
-  endif()
+    list(APPEND DYLYB_ARGS -Xlinker -install_name -Xlinker @rpath/${target}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
-  # Runpath for finding Swift core libraries in the toolchain.
-  # FIXME: Ideally, this should be passed from the swift-ci invocation.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Runpath for finding Swift core libraries in the toolchain.
+    # FIXME: Ideally, this should be passed from the swift-ci invocation.
     list(APPEND DYLYB_ARGS -Xlinker -rpath -Xlinker @loader_path/../../macosx)
-  else()
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Linux)
+    list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN")
     list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN/../../linux")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Windows)
+    # NOTE(compnerd) Windows does not support RPATH
```

Comment:
I'd remove your name from this comment; it makes it look like there's a TODO here, but there isn't - Windows simply doesn't have the concept of rpaths, so there's nothing to do.

Reply:
Done